### ISSUE TITLE
[REQUEST] Multiple Default containers for a single domain

### ISSUE THREAD

**Issue by bobpaul** [mozilla/multi-account-containers#1749]
**Current Behavior**
Currently a user can choose to open a specific site in a container by default. Upon doing so, the user is prompted if they attempt to open that domain in a different container.

**Desired Behavior**
It would be desirable to set multiple "defaults" for a single domain.  When opening in a different container, the message would list all of the "defaults" for that domain.

**Example Use case**
I have a lot several sites that I have multiple accounts for and some of my containers are used to help me manage these multiple logins. While I have ~10 containers, for a given site there might only be 2 or 3 containers that "make sense" to use. 

Example: I have a personal and a business Google account. I would only every like to open Google Drive in my Personal or my Work containers. If I open `drive.google.com` from the default group or click a link to Google drive in a different container, I would like to be presented with the options "Current Container", "Personal Container", and "Work Container".

**Mockup**
<img width="842" alt="image" src="https://user-images.githubusercontent.com/90864/82957761-7326c000-9f79-11ea-8e91-f499c03454d7.png">

**Comment by MichaelKetting** [comment]
I have a similiar scenario with SSO via a company authentication provider. Some of the linked sites are inhouse, some are external applications. I'd like to keep them in separate containers but have them share a common authentication provider / login page.

This might be a duplicate:
https://github.com/mozilla/multi-account-containers/issues/1887

And this might be a related issue / alternative feature request:
https://github.com/mozilla/multi-account-containers/issues/1892

**Comment by securingmom** [comment]
For the SSO affected is a viable temporary workaround multiple Firefox Profiles?

The Profile command line switch can be paired with the no-remote switch to allow both Profiles to run simultaneously 😉

**Comment by holzmaster** [comment]
I have multiple GitHub accounts and I'd always like to open URLs starting with `http://github.com/employeeorganization` in my work container.

**Comment by achernyakevich-sc** [comment]
@bobpaul @holzmaster 

You have touched topic that was in discussion in many other threads. Probably one of the most full summaries I published in [this comment](https://github.com/mozilla/multi-account-containers/issues/2335#issuecomment-1116379147). I hope it will help you.

It look like Containerise add-on would resolve your use case. As well voting for #2294 is very welcome. :)

**Comment by marknelissen** [comment]
@achernyakevich-sc I don't think the comment you link to solves either of the issues brought up by @bobpaul  or @holzmaster . 

The initial problem from @bobpaul is one I'm personally interested in. I would like to indeed assign the exact same domain to multiple containers (google domains) so I can then activate the "only assigned domains" toggle. In the case of multi container assignment, indeed, if coming from outside one of those containers, you should be presented with a choice of which one to use.

The problem from @holzmaster is completely different, and actually a duplicate of https://github.com/mozilla/multi-account-containers/issues/691 which is a broader feature request than https://github.com/mozilla/multi-account-containers/issues/473.

**Comment by ncallaway** [comment]
Since #1892 was closed as a duplicate of this issue, I'll add a note that I have an open PR for #1892 here: https://github.com/mozilla/multi-account-containers/pull/2033.

Note that the implementation is _slightly_ different from the description here (in that its an allow-list instead of truly multiple default containers), so the implementation is closer match to #1892 than this (though they are very similar).

**Comment by dannycolin** [comment]
@ncallaway Thank you for the reminder. I wanted to link the PR to this issue ;). 

As a FYI, I'm working on an new WebExtension API proposal to manage website assignments directly in Firefox instead of in Multi-Account Containers. I'm working on this as a volunteer but hopefully with enough traction from the community we can get this approved upstream. When my proposal will be advanced enough, I'll make sure to share it in this bug (an a few others related to this topic) to get others feedback and make sure we could include as much use cases as it's possible.

**Comment by davidleejy** [comment]
🍻 +1 for this feature request ❤️. The idea that OP @bobpaul raised -- UI buttons to convenience the wielding of multiple containers associated with a single url -- is as relevant as it was three years ago (May 2020).

**Comment by MichaelKetting** [comment]
I find myself more and more blocked by this issue, simply resorting to using private browsing. I'm seriously considering removing the security idea from my thinking and just one or two datagrabbing sites I still need to use on occasion.

**Comment by santo998** [comment]
This FR needs to be implemented.
